### PR TITLE
#7432 issue_fixed

### PR DESCRIPTION
**Pull request by shivamtiwariwebkul**
### Issue  Reference

 #7432 
>The write-review option should not be visible on product card if Allow Guest Review is disabled
>this issue has been resolved

### CODE REVIEW

**Review by Sanjaybhattwebkul — APPROVED** [review]

File: packages/Webkul/Velocity/src/Resources/views/shop/home/new-products.blade.php [hunk 1]
```diff
@@ -1,13 +1,14 @@
 @php
     $count = core()->getConfigData('catalog.products.homepage.no_of_new_product_homepage') ?: 10;
-
+    $guest_review_status = core()->getConfigData('catalog.products.review.guest_review');
     $direction = core()->getCurrentLocale()->direction == 'rtl' ? 'rtl' : 'ltr';
 @endphp
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/home/new-products.blade.php:5]: Move all this logic to the controller. This blade file should be clean and should only have the required variables.

File: packages/Webkul/Velocity/src/Resources/views/shop/home/new-products.blade.php [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
 {!! view_render_event('bagisto.shop.new-products.before') !!}
 
 <product-collections
     count="{{ (int) $count }}"
+    :guest-review-status="{{ (Boolean) $guest_review_status ? 'true' : 'false'}}"
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/home/new-products.blade.php:11]: Be consistent when type casting if above you using a small case then below should also be small case like bool or boolean.

File: packages/Webkul/Velocity/src/Resources/views/shop/home/featured-products.blade.php [hunk 3]
```diff
@@ -1,11 +1,12 @@
 @php
     $count = core()->getConfigData('catalog.products.homepage.no_of_featured_product_homepage') ?: 10;
-
+    $guest_review_status = core()->getConfigData('catalog.products.review.guest_review');
     $direction = core()->getCurrentLocale()->direction == 'rtl' ? 'rtl' : 'ltr';
 @endphp
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/home/featured-products.blade.php:5]: Move all this logic to the controller. This blade file should be clean and should only have the required variables.

File: packages/Webkul/Velocity/src/Resources/assets/js/UI/components/product-card.vue [hunk 4]
```diff
@@ -79,7 +79,7 @@
                 </a>
             </div>
 
-            <div class="product-rating col-12 no-padding" v-else>
+            <div class="product-rating col-12 no-padding" v-else-if="guestReviewStatus">
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/assets/js/UI/components/product-card.vue:82]: All boolean logic should have a `can` or `is` prefix like `canGuestReview`.

Same as `isEnabled`, `isUsed`, `isChecked`, etc.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Velocity/src/Resources/views/shop/home/index.blade.php [hunk 5]
```diff
@@ -63,7 +63,8 @@
         {!! view_render_event('bagisto.shop.home.content.before') !!}
 
             @if ($velocityMetaData)
-                {!! Blade::render($velocityMetaData->home_page_content) !!}
+       
+                {!! Blade::render($velocityMetaData->home_page_content, ['guest_review_status' => $config['guestReviewStatus'], 'count' => $config['count'],'direction' => $config['direction']]) !!}
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/home/index.blade.php:67]: Discussed.

File: packages/Webkul/Shop/src/Http/Controllers/HomeController.php [hunk 6]
```diff
@@ -30,9 +30,15 @@ public function __construct(
...
+        
         $sliderData = $this->sliderRepository->getActiveSliders();
+        $guestReviewStatus = core()->getConfigData('catalog.products.review.guest_review');
+        $count = core()->getConfigData('catalog.products.homepage.no_of_new_product_homepage') ?: 10;
+        $direction = core()->getCurrentLocale()->direction == 'rtl' ? 'rtl' : 'ltr';
 
-        return view($this->_config['view'], compact('sliderData'));
+        $config = ['guestReviewStatus' => $guestReviewStatus , 'count' => $count , 'direction' => $direction];
+
+        return view($this->_config['view'], compact('sliderData','config'));
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/HomeController.php:41]: Discussed.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]
Please do the discussed changes.

**Comment by devansh-webkul** [comment]
I am continuing this.

**Pull request closed without merging**